contracts/contracts/token/OUSD.sol

```diff
@@ -208,229 +220,164 @@ contract OUSD is Initializable, InitializableERC20Detailed, Governable {
         address _from,
         address _to,
         uint256 _value
-    ) public override returns (bool) {
+    ) external returns (bool) {
         require(_to != address(0), "Transfer to zero address");
-        require(_value <= balanceOf(_from), "Transfer greater than balance");
+        require(_value <= allowances[_from][msg.sender], "Allowance exceeded");
```

Comment:
You could cache the allowance otherwise it's read on line 225 and on line 227.

Reply:
good point thanks: https://github.com/OriginProtocol/origin-dollar/pull/2298/commits/d285b3273e4b3a9ca69062f35429647f311be547